pkgs/applications/networking/cluster/terraform-providers/update-provider

```diff
@@ -75,45 +75,44 @@ if [[ -z ${provider} ]]; then
   exit 1
 fi
 
-provider_name=$(basename "${provider}")
-
 # Usage: read_attr <key>
 read_attr() {
-  jq -r ".\"${provider_name}\".\"$1\"" providers.json
+  jq -r ".\"${provider}\".\"$1\"" providers.json
 }
 
 # Usage: update_attr <key> <value>
 update_attr() {
   if [[ $2 == "null" ]]; then
-    jq -S ".\"${provider_name}\".\"$1\" = null" providers.json | sponge providers.json
+    jq -S ".\"${provider}\".\"$1\" = null" providers.json | sponge providers.json
   else
-    jq -S ".\"${provider_name}\".\"$1\" = \"$2\"" providers.json | sponge providers.json
+    jq -S ".\"${provider}\".\"$1\" = \"$2\"" providers.json | sponge providers.json
   fi
 }
 
-prefetch_github() {
-  # of a given owner, repo and rev, fetch the tarball and return the output of
-  # `nix-prefetch-url`
-  local owner=$1
-  local repo=$2
-  local rev=$3
-  nix-prefetch-url --unpack "https://github.com/${owner}/${repo}/archive/${rev}.tar.gz"
+generate_hash() {
+  nix-prefetch -I nixpkgs=../../../../.. \
```

Comment:
a general thought that doesn't have to be addressed: ../ are pretty bad to code re-org as they're hard to grep. It would be nice if there was an env var that points to the repo root.

Reply:
I don't think we have a consistent way of setting environment variables for nixpkgs

Reply:
https://github.com/NixOS/nixpkgs/blob/ad826d3b93b9f2d069beeb3c259d192146309ec3/pkgs/applications/networking/cluster/terraform-providers/update-provider#L2

Needs to be set for `nix-shell` as well. Could probably be handled in there, either `nix-shell` exports the `nixpkgs` that is set manually for the shebang or maybe exports it automatically if it detects that it is in a repo? e.g. `#! nix-shell -I nixpkgs=$REPO_TOP`.

Switched to use `git rev-parse --show-toplevel`.